### ISSUE TITLE
Expiry date feature 

### ISSUE THREAD

**Issue by donjajo** [opensourcepos/opensourcepos#1426]
Hello, 
I'm implementing this in a pharmacy and their drugs has expiry date which they want to get notified when adding them. 
Does this feature exist?

**Comment by SteveIreland** [comment]
That would require a lot tracking feature and that feature isn't currently supported by the database.

**Comment by jekkos** [comment]
You will be able to use attributes in 3.3

**Comment by SteveIreland** [comment]
I'm not sure I want to inject myself in this, but wouldn't this require an expiration date stored at the inventory level?  If so then since attributes is at an item level it doesn't seem an exact fit.

**Comment by daN4cat** [comment]
Yep, also it will require a means to generate alert on data expiration.
More than we have at the moment.
I will reopen for somebody to add the feature one day.

**Comment by daN4cat** [comment]
3.3.0 (now in master) contains the support for datetime attributes. However there is no business logic to trigger an alert past a date.
One alternative is to use an external reporting tool like Reportico searching OSPOS MySQL database for items with close to or past expire date.

**Comment by izeblood** [comment]
Attributes using date CAN'T search... Here is my example:
![expired1](https://user-images.githubusercontent.com/32444665/67664480-ab91fb80-f9a2-11e9-91ba-3c7b5efe23ce.jpg)

then i check search attributes and type the date 10/27/2019 but nothing shows
![expired2](https://user-images.githubusercontent.com/32444665/67664552-d1b79b80-f9a2-11e9-9643-fd97231a1313.jpg)

**Comment by daN4cat** [comment]
If you just type 10, do you get any short list containing the date you are after?
Not sure what the search syntax is in the case of Date in attributes.

I don't think this will be fixed in this release because @jekkos is away until December at least.

**Comment by WebShells** [comment]
Yes, maybe you are using different separators or different search format?

**Comment by daN4cat** [comment]
Tra to search with 2019 only too

**Comment by izeblood** [comment]
> If you just type 10, do you get any short list containing the date you are after?
> Not sure what the search syntax is in the case of Date in attributes.
> 
> I don't think this will be fixed in this release because @jekkos is away until December at least.

If we talk about expiration DATE the format would be like this: MM/DD/YYYY and not all the items/products has the same month or day of expiration but with the same year.My point of view is this, how can we specifically filtered the searching? if we only use Year as our keyword? Days and Months is VERY IMPORTANT things specially when were talking about Pharmacy...it is a very serious matter...

**Comment by daN4cat** [comment]
@izeblood I appreciate but you are not helping here. If in your example you type let say the year in the filter box, do you get a short list with entry containing the date? I just want to make sure the filter works with a partial date. If yes then it's a matter of filter date format.

**Comment by izeblood** [comment]
> @izeblood I appreciate but you are not helping here. If in your example you type let say the year in the filter box, do you get a short list with entry containing the date? I just want to make sure the filter works with a partial date. If yes then it's a matter of filter date format.

NO..... no result even i type the year only which is 2019.

**Comment by daN4cat** [comment]
Thanks 

Try dd-mm-yyyy please

**Comment by daN4cat** [comment]
And yyyy-mm-dd

**Comment by izeblood** [comment]
Woahhhh....  yyyy-mm-dd is working.....
![sample search](https://user-images.githubusercontent.com/32444665/67750063-65ec3600-fa69-11e9-8c8d-79e6cf72871a.jpg)
![search3](https://user-images.githubusercontent.com/32444665/67750100-7dc3ba00-fa69-11e9-80a3-14de4b0b4410.jpg)

**Comment by izeblood** [comment]
![search4](https://user-images.githubusercontent.com/32444665/67750536-5b7e6c00-fa6a-11e9-94f0-dab24d994772.jpg)

**Comment by daN4cat** [comment]
Ok, that's the format date saved in the database.

What's missing is a conversion between date set at UI level and the one in the database.

I leave this issue to be fixed by @jekkos when back and set this for 3.4.0.

For the time being use yyyy-mm-dd.

**Comment by jekkos** [comment]
This is a tricky one: how can we now of a user wants to search for a date here? Try to parse the input as one in the configured format and search on the field it this works?

If we do it this way then we should stick to one date format which will only allow you to search on an exact date (and not a month or year)

**Comment by daN4cat** [comment]
The database format is fixed and consistent across the code. We can't change that, only the format of the conversation of the presentation part can change.

**Comment by jekkos** [comment]
related to #3131
